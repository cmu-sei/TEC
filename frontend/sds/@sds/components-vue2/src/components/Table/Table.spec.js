// TEC - Automated Mismatch Detection Tool
// Copyright 2023 Carnegie Mellon University.
// NO WARRANTY. THIS CARNEGIE MELLON UNIVERSITY AND SOFTWARE ENGINEERING INSTITUTE MATERIAL IS FURNISHED ON AN "AS-IS" BASIS. CARNEGIE MELLON UNIVERSITY MAKES NO WARRANTIES OF ANY KIND, EITHER EXPRESSED OR IMPLIED, AS TO ANY MATTER INCLUDING, BUT NOT LIMITED TO, WARRANTY OF FITNESS FOR PURPOSE OR MERCHANTABILITY, EXCLUSIVITY, OR RESULTS OBTAINED FROM USE OF THE MATERIAL. CARNEGIE MELLON UNIVERSITY DOES NOT MAKE ANY WARRANTY OF ANY KIND WITH RESPECT TO FREEDOM FROM PATENT, TRADEMARK, OR COPYRIGHT INFRINGEMENT.
// Released under a BSD (SEI)-style license, please see license.txt or contact permission@sei.cmu.edu for full terms.
// [DISTRIBUTION STATEMENT A] This material has been approved for public release and unlimited distribution.  Please see Copyright notice for non-US Government use and distribution.
// This Software was created using the SEI Design System and includes and/or makes use of Third-Party Software subject to its own license.
// DM23-003
 
 
import { shallowMount } from '@vue/test-utils'
import Component from "./Table.vue";

describe("Table.vue", () => {
  it("matches snapshot with no props assigned", () => {
    const propsData = {};
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with items and action col props assigned", () => {
    const propsData = {
      items: [
        {
          id: 1,
          name: "A title",
          createdDate: new Date("2000-01-01"),
          lastUpdatedDate: new Date("2014-11-12"),
        },
        {
          id: 2,
          name: "B title",
          createdDate: new Date("2013-02-01"),
          lastUpdatedDate: new Date("2013-10-10"),
        },
      ],
      sortBy: "lastUpdatedDate",
      fields: [
        { key: "name", label: "Title", sortable: true },
        { key: "createdDate", label: "Created", sortable: true, format: (date) => date.toLocaleDateString() },
        {
          key: "lastUpdatedDate",
          label: "Last modified",
          sortable: true, format: (date) => date.toLocaleDateString()
        },
      ],
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with entries but no action col props assigned", () => {
    const propsData = {
      items: [
        {
          id: 1,
          name: "A title",
          createdDate: new Date("2000-01-01"),
          lastUpdatedDate: new Date("2014-11-12"),
        },
        {
          id: 2,
          name: "B title",
          createdDate: new Date("2013-02-01"),
          lastUpdatedDate: new Date("2013-10-10"),
        },
      ],
      fields: [
        { key: "name", label: "Title", sortable: true },
        { key: "createdDate", label: "Created", sortable: true, format: (date) => date.toLocaleDateString() },
        {
          key: "lastUpdatedDate",
          label: "Last modified",
          sortable: true, format: (date) => date.toLocaleDateString()
        },
        { key: "actions", label: "Actions" }
      ]
    };
    const scopedSlots = {
      'cell(actions)': `
        <template #cell(actions)="{item}">
          <button
            class="btn btn-link text-red-300 btn-sm"
            @click="remove(entry)"
          >
            <i class="fas fa-trash"></i>
            Remove entry
          </button>
        </template>
      `,
    };
    const wrapper = shallowMount(Component, { propsData, scopedSlots });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
