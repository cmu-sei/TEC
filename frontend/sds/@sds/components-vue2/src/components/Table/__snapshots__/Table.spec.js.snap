// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.vue matches snapshot with entries but no action col props assigned 1`] = `
"<table data-id=\\"sds-table\\">
  <!---->
  <colgroup>
    <col class=\\"\\">
    <col class=\\"\\">
    <col class=\\"\\">
    <col class=\\"\\">
  </colgroup>
  <thead>
    <tr>
      <th class=\\"space-x-1 select-none group cursor-pointer\\">
        Title
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"space-x-1 select-none group cursor-pointer\\">
        Created
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"space-x-1 select-none group cursor-pointer\\">
        Last modified
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"space-x-1 select-none group\\">
        Actions
        <!---->
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        A title
      </td>
      <td>
        12/31/1999
      </td>
      <td>
        11/11/2014
      </td>
      <td><template></template></td>
    </tr>
    <tr>
      <td>
        B title
      </td>
      <td>
        1/31/2013
      </td>
      <td>
        10/9/2013
      </td>
      <td><template></template></td>
    </tr>
  </tbody>
</table>"
`;

exports[`Table.vue matches snapshot with items and action col props assigned 1`] = `
"<table data-id=\\"sds-table\\">
  <!---->
  <colgroup>
    <col class=\\"\\">
    <col class=\\"\\">
    <col class=\\"null\\">
  </colgroup>
  <thead>
    <tr>
      <th class=\\"space-x-1 select-none group cursor-pointer\\">
        Title
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"space-x-1 select-none group cursor-pointer\\">
        Created
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"space-x-1 select-none group null cursor-pointer\\">
        Last modified
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-100\\">
          <!---->
          <path d=\\"M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z\\" fill=\\"currentColor\\"></path>
          <!---->
        </svg></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        B title
      </td>
      <td>
        1/31/2013
      </td>
      <td>
        10/9/2013
      </td>
    </tr>
    <tr>
      <td>
        A title
      </td>
      <td>
        12/31/1999
      </td>
      <td>
        11/11/2014
      </td>
    </tr>
  </tbody>
</table>"
`;

exports[`Table.vue matches snapshot with no props assigned 1`] = `
"<table data-id=\\"sds-table\\">
  <!---->
  <colgroup></colgroup>
  <thead>
    <tr></tr>
  </thead>
  <tbody></tbody>
</table>"
`;
